### ISSUE TITLE
Prop types and Flow 0.57

### ISSUE THREAD

**Issue by willdurand** [yannickcr/eslint-plugin-react#1487]
Hi

I am upgrading Flow on one of our apps (https://github.com/mozilla/addons-frontend) and get eslint errors related to the `react/prop-types` rule. The following code triggers `react/prop-types` errors for each prop:

```js
type Props = {|
  className?: string,
  width?: number,
  minWidth: number,
  range: number,
|};

export default class LoadingText extends React.Component<Props> {
}
```

```
/Users/william/projects/mozilla/addons-frontend/src/ui/components/LoadingText/index.js
  22:13  error  'className' is missing in props validation  react/prop-types
  22:24  error  'minWidth' is missing in props validation   react/prop-types
  22:34  error  'range' is missing in props validation      react/prop-types
  22:41  error  'width' is missing in props validation      react/prop-types

✖ 4 problems (4 errors, 0 warnings)
```

The code below works though:

```js
type Props = {|
  className?: string,
  width?: number,
  minWidth: number,
  range: number,
|};

export default class LoadingText extends React.Component<Props, void> {
}
```

Any idea? I would like to avoid adding `void` everywhere.. Thanks!

**Comment by willdurand** [comment]
I also have troubles when `State` is defined:

```js
type Props = {|
  className?: string,
  width?: number,
  minWidth: number,
  range: number,
|};

type State = {|
  // attributes
|};

export default class LoadingText extends React.Component<Props, State> {
}
```

Eslint outputs the same problems as mentioned before..

**Comment by willdurand** [comment]
Because of the latest comment, I double checked our dependencies from scratch and one of our dependency did not pull the latest version of this plugin. I am daft, I am very sorry. This works as expected with version `7.4.0`...
